src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2ApplicationNoteEntity.kt

```diff
@@ -19,9 +19,8 @@ data class Cas2ApplicationNoteEntity(
   @Id
   val id: UUID,
 
-  @ManyToOne
-  @JoinColumn(name = "created_by_nomis_user_id")
-  val createdByNomisUser: NomisUserEntity,
+  @Transient
+  final val createdByUser: Cas2User,
```

Comment:
this seemed a good way to ensure that the Entity will always define either `createdByNomisUser` or `createdByExternalUser`... if a new type of CAS2User that is not able to create a note was added, it could throw an error in the `when` block below.

Reply:
I'm 50/50 on whether this should be done here (as you've done) or in thing building the Entity but I can't strongly argue it either way!

I like how both the constructor and getCreatedByNomisOrExternalUser deal with your Cas2User interface, hiding away the internal implementation details.

My only suggestion would be to pull the new init block up just below the constructor because it can be missed at the moment.

Use of @Transient looks good - coming from Java i'd normally always avoid @Transients but in this context i think it's perfect - you're telling JPA that this is 'just' a constructor arg and shouldn't be directly persisted into a column

Reply:
nice feedback, thank you!
> My only suggestion would be to pull the new init block up just below the constructor because it can be missed at the moment. 

I think it's not possible to put it above the `createdByNomisUser` and `createdByExternalUser` var's, as they need to be declared before the `init` function, hopefully as it's the first of the listed functions it should be clear enough?

Reply:
Yep you're right, it's as high as it can be 👍